packages/ra-ui-materialui/src/button/PrevNextButton.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import {
+    useCreatePath,
+    useGetList,
+    useGetRecordId,
+    useListParams,
+    useResourceContext,
+    useTranslate,
+} from 'ra-core';
+import { NavigateBefore, NavigateNext } from '@mui/icons-material';
+import { Link } from 'react-router-dom';
+import { IconButton } from '@mui/material';
+
+export const PrevNextButton = (props: PrevNextButtonProps) => {
+    const { linkType = 'edit' } = props;
+    const translate = useTranslate();
+    const recordId = useGetRecordId();
+    const resource = useResourceContext();
+    const createPath = useCreatePath();
+    const [query] = useListParams({
+        resource,
+    });
```

Comment:
`useListParams` does more than you need: it merges params from the URL, the store, and the props. You only need the last two, so you must do the merge manually.

Reply:
So I guess I need to remove `useListParams` in favor of `useStore` and probably another hook?

Reply:
exactly